src/rustllvm/ArchiveWrapper.cpp

```diff
@@ -149,7 +149,20 @@ LLVMRustArchiveIteratorNext(LLVMRustArchiveIteratorRef rai) {
 #endif
     Archive::Child *ret = new Archive::Child(child);
 
+    // This sets rai->err (in recent LLVM). Even if no error occurs,
+    // we need to at least check the Error object, otherwise LLVM calls abort().
     ++rai->cur;
+    // It would probably be better to communicate this error to the caller
+    // while also returning the child we successfully retrieved above, but
+    // that complicates the API.
+    // As a broken archive is bad news anyway, let's just return NULL
+    // "one child too soon".
+    #ifdef LLVM_VERSION_GE(3, 9)
```

Comment:
I think this should be `#if`, right?

Also, if this branch is taken, that should free `ret`, right?